core/src/main/scala/com/velocidi/apso/json/ExtraJsonProtocol.scala

```diff
@@ -170,6 +171,15 @@ trait ExtraMiscJsonProtocol {
   implicit val localDateDecoder: Decoder[LocalDate] =
     Decoder[String].emapTry(v => Try(new LocalDate(v)))
 
+  implicit def currencyDecoder(implicit moneyContext: MoneyContext): Decoder[Currency] =
+    Decoder[String].emap { code =>
+      // TODO: this can be improved when https://github.com/typelevel/squants/pull/330 is accepted and released
+      val currencyOpt = moneyContext.currencies.find(_.code == code)
```

Comment:
We should be able to use the `asCurrency` method here.

Reply:
Done in 0097f80010a3564b96f213fb901834d9c8dce153